proxy/proxy.go

```diff
@@ -555,13 +567,52 @@ func (p *Proxy) checkBreaker(c *context) (func(bool), bool) {
 	return done, ok
 }
 
+func ratelimitError(settings ratelimit.Settings) error {
+	return &proxyError{
+		err:  errRatelimitError,
+		code: http.StatusTooManyRequests,
+		additionalHeader: http.Header{
+			ratelimit.Header: []string{strconv.Itoa(settings.MaxHits * int(time.Hour/settings.TimeWindow))},
+		},
+	}
+}
+
+// TODO(sszuecs) not in use for the global rate limiter, but should be
+// used in per route rate limiting
+func (p *Proxy) checkRatelimit(c *context) bool {
```

Comment:
this function is not called now, let's delete it, and document the idea rather next to the Lookuper.